packages/core/content-manager/admin/src/pages/EditView/components/DocumentActions.tsx

```diff
@@ -683,11 +725,108 @@ const PublishAction: DocumentActionComponent = ({
+            params: currentDocumentMeta.params,
+          };
+
+          /*
+           * Update, if needed, the parent relation with the newly published document.
+           * Check if in history we have the parent relation otherwise use the
+           * rootDocument
+           */
+          if (
+            fieldToConnect &&
+            documentHistory &&
+            (parentDocumentMetaToUpdate.documentId ||
+              parentDocumentMetaToUpdate.collectionType === SINGLE_TYPES)
+          ) {
+            const parentDataToUpdate =
+              parentDocumentMetaToUpdate.collectionType === SINGLE_TYPES
+                ? getInitialFormValues()
+                : parentDocumentData.getInitialFormValues();
+            const metaDocumentToUpdate = documentHistory.at(-2) ?? rootDocumentMeta;
+
+            /*
+             * Check if the fieldToConnect is already present in the parentDataToUpdate.
+             * This happens in particular when in the parentDocument you have created
+             * a new component without saving.
+             */
+            const isFieldPresent = !!get(parentDataToUpdate, fieldToConnect);
+            const fieldToConnectPath = isFieldPresent
+              ? fieldToConnect
+              : // Compute the path to the parent object
+                fieldToConnect.split('.').slice(0, -1).join('.');
+            const fieldToConnectValue = isFieldPresent
+              ? {
+                  connect: [
+                    {
+                      id: res.data.documentId,
+                      documentId: res.data.documentId,
+                      locale: res.data.locale,
+                    },
+                  ],
+                }
+              : {
+                  [fieldToConnect.split('.').pop()!]: {
+                    connect: [
+                      {
+                        id: res.data.documentId,
+                        documentId: res.data.documentId,
+                        locale: res.data.locale,
+                      },
+                    ],
+                    disconnect: [],
+                  },
+                  __component: fieldToConnectUID,
+                };
+            // In case the object was not present you need to pass the componentUID of the parent document
+            const objectToConnect = set({}, fieldToConnectPath, fieldToConnectValue);
+            const dataToUpdate = merge(parentDataToUpdate, objectToConnect);
+
+            try {
+              const updateRes = await updateDocumentMutation({
+                collectionType: metaDocumentToUpdate.collectionType,
+                model: metaDocumentToUpdate.model,
+                documentId:
+                  metaDocumentToUpdate.collectionType !== SINGLE_TYPES
+                    ? metaDocumentToUpdate.documentId
+                    : undefined,
+                params: metaDocumentToUpdate.params,
+                data: {
+                  ...dataToUpdate,
+                },
```

Comment:
I don't think the spread is necessary:

```suggestion
                data: dataToUpdate,
```